skyportal/handlers/api/internal/plot.py

```diff
@@ -30,14 +30,19 @@ def get(self, obj_id):
         # Just return browser by default if not one of accepted types
         if device not in device_types:
             device = "browser"
-        json = plot.photometry_plot(
-            obj_id,
-            self.current_user,
-            width=int(width),
-            device=device,
-        )
-        self.verify_and_commit()
-        self.success(data={'bokehJSON': json, 'url': self.request.uri})
+        with self.Session() as session:
+            try:
+                json = plot.photometry_plot(
+                    obj_id,
+                    self.current_user.id,
+                    session,
+                    width=int(width),
+                    device=device,
```

Comment:
Both here and in the spectroscopy plot: 
I would explicitly name each parameter I'm sending into the functions. 
There are starting to be a lot of them, and the order is hard to guess.

Reply:
Good idea!